aws-samples/serverless-patterns
# New serverless pattern - AWS Transfer -> S3

*Description of changes:*

# AWS Transfer SFTP-> S3

This pattern in CDK offers an example to setup an AWS SFTP server using [AWS Transfer for SFTP](https://aws.amazon.com/aws-transfer-family/). You can simply create a server, set up user accounts, and associate the server with one or more Amazon Simple Storage Service (Amazon S3) buckets.
You have fine-grained control over user identity, permissions, and keys. You can create users within Transfer for SFTP. You can also use IAM policies to control the level of access granted to each user.

Learn more about this pattern at Serverless Land Patterns: https://serverlessland.com/patterns/cdk-awstransfer.

Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.


![image](https://user-images.githubusercontent.com/93898993/153142722-7ad3fa35-dfef-4d45-a5a1-ff16b8cb8bf7.png)

Learn more about this pattern at Serverless Land Patterns: https://serverlessland.com/patterns/cdk-awstransfer.

This example creates AWS SFTP server which is accessible over internet. User along public key and fine grained access to S3 bucket will be created.

## Testing

1. Deploy the app, by specifying account and region within configurations.
2. Switch to AWS Transfer services on the AWS console. Switch to your region, you will see SFTP Server created.
2. Click on the server, you will find the user which its default HomeDirectory.
3. Connect to you server from your terminal or windows ssh client `sftp -i {transfer-key-private-key} {serverid}.server.transfer.{region}.amazonaws.com`.

## Cleanup

1. Delete the stack
    ```bash
    aws cloudformation delete-stack --stack-name STACK_NAME
    ```
1. Confirm the stack has been deleted
    ```bash
    aws cloudformation list-stacks --query "StackSummaries[?contains(StackName,'STACK_NAME')].StackStatus"
    ```
1. ```bash
    cdk destroy
   ```
